hoot-core/src/main/cpp/hoot/core/util/LibPostalInit.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2017, 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#include "LibPostalInit.h"
+
+// Hoot
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/HootException.h>
+
+// libpostal
+#include <libpostal/libpostal.h>
+
+namespace hoot
+{
+
+LibPostalInitPtr LibPostalInit::_theInstance;
+
+LibPostalInit::LibPostalInit()
+{
+  // This init takes ~5 seconds normally.
+  if (!libpostal_setup_datadir(ConfigOptions().getLibpostalDataDir().toUtf8().data()) ||
+      !libpostal_setup_parser_datadir(ConfigOptions().getLibpostalDataDir().toUtf8().data()) ||
+      !libpostal_setup_language_classifier_datadir(
+        ConfigOptions().getLibpostalDataDir().toUtf8().data()))
+  {
+    throw HootException("libpostal setup failed.");
+  }
+}
+
```

Comment:
Its worth noting that this singleton init of libpostal is fairly expensive, I'm guessing due to some of the address parsing related data it needs to load into memory.  For regular runtime, not that big of deal when running a conflation job.  When running tests, though, since this penalty is paid per hoot process, can become a big deal.  I went through all the conflate related tests and disabled poi/poly address conflation (poi.polygon.address.match.enabled=false) on any tests running all the matchers that didn't really need it to get test times back down to normal.